src/index.ts

```diff
@@ -18,6 +26,25 @@ export interface JsonFetchOptions extends Omit<RequestInit, 'body'> {
   retry?: Parameters<typeof promiseRetry>[0];
   timeout?: number;
   expectedStatuses?: Array<number>;
+  onRequestStart?: ({
+    agent,
+    body,
+    retry,
+    timeout,
+    expectedStatuses,
+    url,
+    retryCount,
+  }: JsonFetchOptions & OnRequestOptions) => void;
+  onRequestEnd?: ({
+    agent,
+    body,
+    retry,
+    timeout,
+    expectedStatuses,
+    responseOrError,
+    url,
+    retryCount,
+  }: JsonFetchOptions & OnRequestOptions & OnRequestEndOptions) => void;
```

Comment:
🤔 since this is just a type definition, do we have to destructure?
```suggestion
  onRequestStart?: (opts: JsonFetchOptions & OnRequestOptions) => void;
  onRequestEnd?: (opts: JsonFetchOptions & OnRequestOptions & OnRequestEndOptions) => void;
```